admin/javascript/src/components/grid-field/table.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import SilverStripeComponent from 'silverstripe-component';
+
+class GridFieldTableComponent extends SilverStripeComponent {
```

Comment:
A "grid field" is more than just a table, there's button rows, etc. Also, `GridField` should really be called `GridFieldContainer` - it's the "smart" component which hooks into Redux state management, and instanciates a "dumb" `GridFieldTableComponent`.

I wouldn't overthink the current component structure, my hope is that we'll replace this with a third party component like  https://github.com/GriddleGriddle/Griddle (there's even a PoC for SS already: https://github.com/Tom-Alexander/silverstripe-griddle)